kinesis-lambda-error-handling/README.md

```diff
@@ -0,0 +1,108 @@
+
+## Deployment Instructions
+
+1. Clone the project to your local working directory
+
+   ```sh
+   git clone https://github.com/aws-samples/serverless-patterns/ 
+   ```
+
+2. Change the working directory to this pattern's directory
+
+   ```sh
+   cd serverless-patterns/kinesis-lambda-error-handling
+   ```
+3. From the command line, use AWS SAM to build and deploy the AWS resources for the pattern as specified in the template.yml file:
+   ```
+   sam build
+   sam deploy --guided
+   ```
+4. During the prompts:
+
+   - Enter a stack name
+   - Enter the desired AWS Region
+   - Allow SAM CLI to create IAM roles with the required permissions.
+
+   Once you have run `sam deploy --guided` mode and saved arguments to a configuration file (samconfig.toml), you can use `sam deploy` in the future to use these defaults.
+   Note the outputs from the SAM deployment process. These contain the resource names and/or ARNs which are used for testing.
+
+## How it works
+![Reference Architecture](/images/kinesis-lambda-error-handling.jpg)
+
+The pattern builds infrastructure with Amazon Kinesis data stream, a consumer Lambda function, SQS queue to load the failed records for further troubleshooting, and cloudwatch to validate the logs for the success, failure with retries.
+
+The event source mapping that reads records from your Kinesis stream, invokes AWS Lambda function synchronously, and retries on errors. If Lambda throttles the function or returns an error without invoking the function, Lambda retries until the records expire or exceed the maximum age that you configure on the event source mapping.
+
+## Testing 
+To test the pattern, we are breaking the test cases into two scenarios.
+
+### Scenario 1: Put messages without the Poision pill
+- test-records-without-poison-pill.json --> Holds the base64 data with partition key
+- without-poison-pill-put-records.sh --> bash script to put records into Kinesis data stream
+
+```
+chmod +x kinesis-producer/*.sh;
+./kinesis-producer/without-poison-pill-put-records.sh
+```
+
+- Navigate to AWS Console, and then to Cloudwatch, Log groups, select the log group and the latest Log stream
+- In the logs, you should see all the 3 messages processed without any exception. 
+
+### Scenario 2: Put messages with the Poision pill
+- test-records-with-poison-pill.json --> Holds the base64 data with partition key
+- with-poison-pill-put-records.sh --> bash script to put records into Kinesis data stream
+
+```
+./kinesis-producer/with-poison-pill-put-records.sh
+```
+
+- Cloudwatch logs shows that there was one invalid or poison message, so here the bisectonfailure was applied which lead Lambda to split the batch in half and resume each half separately. Maximum retry attempts and maximum record age limit the number of retries on a failed batch. As the retry limit is 5, it will retry 5 times, before the message is put to AWS SQS.
```

Comment:
same here, use sam logs to get the logs

Reply:
also if you can add CLI instructions to check the DLQ for the message it would be great as well.